Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java

```diff
@@ -614,17 +614,81 @@ private static DocStruct getOrCreateChild(DocStruct parent, String type, String
 		try {
 			return parent.getChild(type, identifierField, identifier);
 		} catch (NoSuchElementException nose) {
-			DocStruct child = parent.createChild(type, act, ruleset);
+			DocStruct child = act.createDocStruct(ruleset.getDocStrctTypeByName(type));
 			child.addMetadata(identifierField, identifier);
 			try {
 				child.addMetadata(optionalField, identifier);
 			} catch (MetadataTypeNotAllowedException e) {
+				logger.warn(e.getMessage());
```

Comment:
`logger.isEnabled` does only make sense if the logging code does something, like `String` composition. Here the work has already been done so there is nothing worth skipping.